### DISCUSSION TITLE
Error in upgrade.c file

### DISCUSSION THREAD

**Discussion by altairscorpio** [micronucleus/micronucleus#257] [Q&A]
Hey everyone! i tried to compile hex but this happening, main.hex was compiled after that but in git bash i getting those errors

`$ make`
`upgrade.c: In function 'write_new_bootloader':`
`upgrade.c:94:23: error: 'bootloader_size' undeclared (first use in this function)`
`   while ( page_addr < bootloader_size ) {`
`                       ^`
`upgrade.c:94:23: note: each undeclared identifier is reported only once for each function it appears in`
`upgrade.c:98:32: error: 'bootloader' undeclared (first use in this function)`
`       int subaddress = ( (int) bootloader ) + page_addr + word_addr;`
`                                ^`
`upgrade.c:99:33: error: 'bootloader_end' undeclared (first use in this function)`
`       if ( subaddress < ( (int) bootloader_end ) ) { // valid boot code`
`                                 ^`
`upgrade.c:107:17: error: 'bootloader_address' undeclared (first use in this function)`
`     erase_page( bootloader_address + page_addr );`
`                 ^`
`upgrade.c: In function 'forward_interrupt_vector_table':`
`upgrade.c:153:39: error: 'bootloader_address' undeclared (first use in this function)`
`     vector_table[ iter ] = 0xC000 + ( bootloader_address / 2 ) - 1;`
`                                       ^`
`make: *** [upgrade.o] Error 1`

**Comment by cpldcpu** [comment]
This is rather strange, these variables are defines in the makefile, why should that fail? It's a bit tough to debug this remotely, my guess si something with the toolchain is off.

Have you tried the AVR-GCC in the atmel studio shell? that does usually work.

Do you need the upgrade tool? Otherwise maybe it is ok just to program the main firmware hex and ignore that error.
